src/background/background-message-distributor.ts

```diff
@@ -43,6 +43,8 @@ export class BackgroundMessageDistributor {
 
         if (response) {
             return response;
+        } else {
+            return Promise.resolve();
```

Comment:
This is going to return `Promise.resolve()` for all `interpreter` paths. I don't think that's what we want to do, since that's masking the fact that those paths are all fire-and-forget cases we haven't yet async-ified.

Instead, I think it would be preferable to only invent a `Promise.resolve()` for the case where we don't know how to interpret it at all (L41), and leave the interpreter paths returning `undefined` until we come up with a mechanism to enable starting to make them async.